executor/show.go

```diff
@@ -141,15 +167,24 @@ func (e *ShowExec) fetchAll() error {
 }
 
 func (e *ShowExec) fetchShowEngines() error {
-	row := types.MakeDatums(
-		"InnoDB",
-		"DEFAULT",
-		"Supports transactions, row-level locking, and foreign keys",
-		"YES",
-		"YES",
-		"YES",
-	)
-	e.rows = append(e.rows, row)
+	if e.forChunk {
+		e.result.AppendString(0, "InnoDB")
+		e.result.AppendString(1, "DEFAULT")
+		e.result.AppendString(2, "Supports transactions, row-level locking, and foreign keys")
+		e.result.AppendString(3, "YES")
+		e.result.AppendString(4, "YES")
+		e.result.AppendString(5, "YES")
```

Comment:
Can we implement something like `MakeDatums` so that we could save lots of repeated work?